Don't eagerly create AliasInfo in parseAliasDeclaration

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151682
* #151697
* __->__ #151630
* #151629
* #151628
* #151627
* #151626

No need to create an AliasInfo...unless we need it.

Differential Revision: [D73129452](https://our.internmc.facebook.com/intern/diff/D73129452/)

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

swolchok committed:
Optimize parseAliasDeclaration

No need to create an AliasInfo...unless we need it.

Differential Revision: [D73129452](https://our.internmc.facebook.com/intern/diff/D73129452/)

[ghstack-poisoned]

Skylion007 commented on torch/csrc/jit/frontend/schema_type_parser.cpp:
```diff
@@ -153,15 +153,16 @@ std::optional<AliasInfo> SchemaTypeParser::parseAliasAnnotation() {
   } else if (L.nextIf('!')) {
+    AliasInfo alias_info;
     alias_info.addBeforeSet(
         Symbol::fromQualString("alias::$" + std::to_string(next_id++)));
```
Fmtlib could be used here.

swolchok replied:
true, but outside the intended scope of the PR. I've updated the title to make that more clear.

malfet reviewed (approved):
LGTM

swolchok committed:
Update on "Don't eagerly create AliasInfo in parseAliasDeclaration"


No need to create an AliasInfo...unless we need it.

Differential Revision: [D73129452](https://our.internmc.facebook.com/intern/diff/D73129452/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
